plasma_framework/contracts/src/exits/payment/controllers/PaymentProcessInFlightExit.sol

```diff
@@ -133,14 +138,14 @@ library PaymentProcessInFlightExit {
     {
         uint256 inputNumOfTheToken;
         for (uint i = 0; i < MAX_INPUT_NUM; i++) {
-            if (exit.inputs[i].token == token) {
+            if (exit.inputs[i].token == token && exit.inputs[i].amount > 0) {
                 inputNumOfTheToken++;
             }
         }
         bytes32[] memory outputIdsOfInputs = new bytes32[](inputNumOfTheToken);
         uint sameTokenIndex = 0;
         for (uint i = 0; i < MAX_INPUT_NUM; i++) {
-            if (exit.inputs[i].token == token) {
+            if (exit.inputs[i].token == token && exit.inputs[i].amount > 0) {
```

Comment:
Why was that added check necessary?

Reply:
I want to be consistent. We flag only when the input is not empty however we check inputs even if it is empty. It only works because empty input --> outputId == 0 --> the flag of that would always be false. I think it is better to not check the flag for that.

Reply:
Also, I abstract the check of `amount > 0` to be more explicit checking whether it is an empty withdraw data and have a function in library for that.